controller/user.js

```diff
@@ -128,4 +130,92 @@ const updateAvatarUser = async (file, userId) => {
   }
 };
 
-module.exports = { login, current, signup, logout, updateAvatarUser };
+const verifyUser = async (verificationToken) => {
+  try {
+    const user = await User.findOne({ verificationToken });
+
+    if (!user) {
+      throw new Error("User not found");
+    }
+
+    user.verificationToken = null;
+    user.verify = true;
+    await user.save();
+
+    return { message: "Verification successful" };
+  } catch (error) {
+    console.error("Verify User Error:", error.message);
+    throw error;
+  }
+};
+
+const sendVerificationEmail = async (email, verificationToken) => {
+  try {
+    const transporter = nodemailer.createTransport({
+      service: "gmail",
+      auth: {
+        user: process.env.EMAIL_USER,
+        pass: process.env.EMAIL_PASSWORD,
+      },
+    });
+
+    const mailOptions = {
+      from: process.env.EMAIL_USER,
+      to: email,
+      subject: "Verification Email",
+      text: `Click the following link to verify your email: http://your_api_base_url/users/verify/${verificationToken}`,
```

Comment:
```suggestion
      text: `Click the following link to verify your email: http://localhost:3000/users/verify/${verificationToken}`,
```
Tu gdzie jest ta zmiana to musisz podać twój prawdziwy port. Normalnie to to byłby cały adres do twojego serwera.

Reply:
Rozumiem że jeszcze tego nie testowałaś, prawda?